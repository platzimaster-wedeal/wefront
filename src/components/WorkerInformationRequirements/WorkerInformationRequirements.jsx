import React from "react";

// Styles
import "../../assets/styles/components/WorkerInformationRequirements/WorkerInformationRequirements.scss";

// Components
import Requirements from "../Requirements/Requirements";

const WorkerInformationRequirements = ({ requirements }) => {
 return (
  <div className="worker-information-requirements">
   <strong className="worker-information-requirements__title">
    Requirements
   </strong>
   <ul className="worker-information-requirements__list">
    {requirements && requirements.length > 0 ? (
     requirements.map((req, i) => (
      <li key={i}>
       <Requirements title={req} />
      </li>
     ))
    ) : (
     <Requirements title="There is no requirements!" />
    )}
   </ul>
  </div>
 );
};

export default WorkerInformationRequirements;
