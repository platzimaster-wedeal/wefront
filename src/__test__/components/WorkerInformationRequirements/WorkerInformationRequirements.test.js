import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import WorkerInformationRequirements from "../../../components/WorkerInformationRequirements/WorkerInformationRequirements";

describe("Test for the component WorkerInformationRequirements :D", () => {
 const WorkerInformationRequirementsMounted = mount(
  <WorkerInformationRequirements requirements={["new requirement"]} />
 );
 const WorkerInformationRequirementsCreated = create(
  <WorkerInformationRequirements />
 );

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(WorkerInformationRequirementsCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(WorkerInformationRequirementsMounted.length).toEqual(1);
  });
  test("Should the title of the component be a string", () => {
   expect(
    typeof WorkerInformationRequirementsMounted.find(
     ".worker-information-requirements__title"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the component equal requirements", () => {
   expect(
    WorkerInformationRequirementsMounted.find(
     ".worker-information-requirements__title"
    ).text()
   ).toEqual("Requirements");
  });
 });
});
