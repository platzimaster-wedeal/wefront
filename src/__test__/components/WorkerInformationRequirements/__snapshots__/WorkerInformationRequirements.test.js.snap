// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component WorkerInformationRequirements :D Visual tests for the component should the comopnent have the same tree 1`] = `
<div
  className="worker-information-requirements"
>
  <strong
    className="worker-information-requirements__title"
  >
    Requirements
  </strong>
  <ul
    className="worker-information-requirements__list"
  >
    <div
      className="requirements__container"
    >
      <svg
        className="requirements__container-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
        />
      </svg>
      <p
        className="requirements__container-content"
      >
        There is no requirements!
      </p>
    </div>
  </ul>
</div>
`;
